components/brave_vpn/brave_vpn_connection_manager_mac.mm

```diff
@@ -0,0 +1,249 @@
+      setLifetimeMinutes:1440];  // 24 hours
+  [[protocol_config childSecurityAssociationParameters]
+      setEncryptionAlgorithm:NEVPNIKEv2EncryptionAlgorithmAES256GCM];
+  [[protocol_config childSecurityAssociationParameters]
+      setDiffieHellmanGroup:NEVPNIKEv2DiffieHellmanGroup20];
+  [[protocol_config childSecurityAssociationParameters]
+      setLifetimeMinutes:480];  // 8 hours
+
+  return protocol_config;
+}
+
+}  // namespace
+
+// static
+BraveVPNConnectionManager* BraveVPNConnectionManager::GetInstance() {
+  static base::NoDestructor<BraveVPNConnectionManagerMac> s_manager;
+  return s_manager.get();
+}
+
+BraveVPNConnectionManagerMac::BraveVPNConnectionManagerMac() = default;
+BraveVPNConnectionManagerMac::~BraveVPNConnectionManagerMac() = default;
+
+void BraveVPNConnectionManagerMac::CreateVPNConnection(
+    const BraveVPNConnectionInfo& info) {
+  info_ = info;
+
+  if (StorePassword([NSString stringWithUTF8String:info_.password().c_str()],
+                    kBraveVPNKey) != errSecSuccess)
+    return;
+
+  NEVPNManager* vpn_manager = [NEVPNManager sharedManager];
+  [vpn_manager loadFromPreferencesWithCompletionHandler:^(NSError* error) {
+    if (error) {
+      LOG(ERROR) << "Create - loadFromPrefs error: "
+                 << base::SysNSStringToUTF8([error localizedDescription]);
+      return;
+    }
+
+    [vpn_manager setEnabled:YES];
+    [vpn_manager setProtocolConfiguration:CreateProtocolConfig(info_)];
+    [vpn_manager setLocalizedDescription:base::SysUTF8ToNSString(
+                                             info_.connection_name())];
+
+    [vpn_manager saveToPreferencesWithCompletionHandler:^(NSError* error) {
+      if (error) {
+        LOG(ERROR) << "Create - saveToPrefs error: "
+                   << base::SysNSStringToUTF8([error localizedDescription]);
+        return;
+      }
+      VLOG(2) << "Create - saveToPrefs success";
+      for (Observer& obs : observers_)
+        obs.OnCreated(std::string());
+    }];
+  }];
+}
+
+void BraveVPNConnectionManagerMac::UpdateVPNConnection(
+    const BraveVPNConnectionInfo& info) {
+  NOTIMPLEMENTED();
+}
+
+void BraveVPNConnectionManagerMac::RemoveVPNConnection(
+    const std::string& name) {
+  NEVPNManager* vpn_manager = [NEVPNManager sharedManager];
+  [vpn_manager loadFromPreferencesWithCompletionHandler:^(NSError* error) {
+    if (error) {
+      LOG(ERROR) << "RemoveVPNConnection - loadFromPrefs: "
+                 << base::SysNSStringToUTF8([error localizedDescription]);
+    } else {
+      [vpn_manager removeFromPreferencesWithCompletionHandler:^(
+                       NSError* error) {
+        if (error) {
+          LOG(ERROR) << "RemoveVPNConnection - removeFromPrefs: "
+                     << base::SysNSStringToUTF8([error localizedDescription]);
+        }
+        VLOG(2) << "RemoveVPNConnection - successfully removed";
+        for (Observer& obs : observers_)
+          obs.OnRemoved(std::string());
```

Comment:
This may be called on any thread?

Reply:
I expected this is called on main thread because docs says like below.
https://developer.apple.com/documentation/networkextension/nevpnmanager/1406177-loadfrompreferenceswithcompletio?language=objc
```
This block will be executed on the caller's main thread after the load operation is complete
```